src/App/Application.cpp

```diff
@@ -2053,6 +2055,12 @@ void Application::ParseOptions(int ac, char ** av)
             Base::Interpreter().addPythonPath(It->c_str());
     }
 
+    if (vm.count("python-home")) {
+        string py_home = vm["python-home"].as<string>();
+        wchar_t * py_home_w_char =  const_cast<wchar_t*>(boost::locale::conv::utf_to_utf<wchar_t>(py_home).c_str());

```

Comment:
This line is only needed for Py3. For Py2 you directly pass py_home to the function.
About boost::locale::conv::utf_to_utf I don't know if this right function. py_home is utf-8 encoded and the output should be utf-16.  At least in InterpreterSingleton::init we used Py_DecodeLocale and _Py_char2wchar.

Furthermore I am not sure if calling Py_SetPythonHome does something useful if the Python interpreter is already initialized.